js/web/lib/wasm/jsep/webgpu/ops/binary-op.ts

```diff
@@ -173,15 +171,30 @@ const createBinaryOpProgramInfo =
         vectorize = true;
       }
 
+      const useShapesUniforms = isBroadcast && enableShapesUniforms(a.dims.length) &&
+          enableShapesUniforms(b.dims.length) && enableShapesUniforms(outputShape.length);
       return {
         name,
-        shaderCache: {hint: cacheKey},
+        shaderCache: {
+          hint: cacheKey,
```

Comment:
It seems that `vectorize`, `isAOneElement`, `isBOneElement` should also be part of `cacheKey`. Otherwise, how do you distinguish the different path in shader?

Reply:
Added `vectorize, isAOneElement, isBOneElement and isBroadcast` to `cacheKey`